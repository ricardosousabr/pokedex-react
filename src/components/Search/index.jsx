import { useState } from "react";
import axios from "axios";
import {
  Container,
  Title,
  InformationPokemon,
  NamePoke,
  ItemList,
  Button,
  SearchInput,
  BoxSearch,
  BarStatus,
} from "./styles";
import States from "../States/index";
export default function Search() {
  const [pokemonData, setPokemonData] = useState("");
  const [pokemonName, setPokemonName] = useState("");
  const apiPokemon = "https://pokeapi.co/api/v2/pokemon/" + pokemonName;

  function getPokemon() {
    axios
      .get(apiPokemon)
      .then((response) => {
        setPokemonData(response.data);
        console.log(response.data);
      })
      .catch((error) => console.log(error));
  }

  function clearInputpokemon() {
    setPokemonName("");
  }

  function handleSubmit(e) {
    e.preventDefault();

    getPokemon();
    clearInputpokemon();
  }

  return (
    <Container>
      <div>
        <Title>Pokedex</Title>
      </div>
      <div>
        <BoxSearch onSubmit={handleSubmit}>
          <SearchInput
            placeholder="Digite o nome do pokemon"
            type="text"
            value={pokemonName}
            onChange={(e) => setPokemonName(e.target.value)}
          />
          <Button>Search</Button>
        </BoxSearch>
      </div>
      <InformationPokemon>
        <ul>
          {pokemonData ? (
            <ItemList>
              <div>
                <NamePoke>{pokemonData.name}</NamePoke>
              </div>
              <div>
                <img
                  src={pokemonData.sprites.front_default}
                  alt=""
                  width={100}
                  height={100}
                />
              </div>
              <div>
                <States statesPokemon={pokemonData} />
              </div>
            </ItemList>
          ) : (
            <li>Escreva o nome do seu pokemon</li>
          )}
        </ul>
      </InformationPokemon>
    </Container>
  );

  module.exports = getPokemon;
  module.exports = clearInputpokemon;
}
