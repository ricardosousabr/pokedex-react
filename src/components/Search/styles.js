import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: 100vh;
`;

export const Title = styled.h1`
 color: white;
 padding-bottom: 35px;
`;

export const NamePoke = styled.h2`
 color: white;
 padding: 35px 0;
 text-align: center;
`;

export const ItemList = styled.li`
  list-style: none;
`;

export const InformationPokemon = styled.div`
  color: white;
`;

export const Button = styled.button`
  background: #6f6799;
  color: white;
  width: 90px;
  border: none;
  height: 30px;
  border-radius: 14px;
`;

export const BoxSearch = styled.form`
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 350px;

`;

export const SearchInput = styled.input`
  width: 200px;
  height: 30px;
  color: white;
  background: transparent;
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: 1px solid;
  :focus {
    border-top: none;
    border-right: none;
    border-left: none;
    border-bottom: 1px solid;
    outline: 0;
  }
`;

export const BarStatus = styled.progress`
  height: 10px;
  border-radius: 10px;

  ::-webkit-progress-value {
    background-color: red;
  }
`;