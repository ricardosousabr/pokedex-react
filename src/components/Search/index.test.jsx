/**
 * @jest-environment jsdom
 */

import React from 'react'
import {render} from '@testing-library/react'
import Search from './index'

const getPokemon = require('./index')

it("renders homepage unchanged", () => {
  const { container } = render(<Search />)
  expect(container).toMatchSnapshot()
});
