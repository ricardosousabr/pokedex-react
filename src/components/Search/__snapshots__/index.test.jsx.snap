// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="styles__Container-sc-dxxv4y-0 gGfPao"
  >
    <div
      class="styles__ContainerImage-sc-dxxv4y-1 eSUxBk"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Fimg%2Fpokemon-logo.png&w=828&q=75"
        srcset="/_next/image?url=%2Fimg%2Fpokemon-logo.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fpokemon-logo.png&w=828&q=75 2x"
        style="color: transparent;"
        width="400"
      />
    </div>
    <div
      class="styles__ContainerForm-sc-dxxv4y-2 gKIHLC"
    >
      <form
        class="styles__BoxSearch-sc-dxxv4y-7 glGlCx"
      >
        <input
          class="styles__SearchInput-sc-dxxv4y-8 gFjLP"
          placeholder="Digite o nome do pokemon"
          type="text"
          value=""
        />
        <button
          class="styles__Button-sc-dxxv4y-6 isOtJA"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="styles__ContainerPokemon-sc-dxxv4y-3 clmjgs"
    >
      <ul
        class="styles__ListSkils-sc-dxxv4y-5 epXqgX"
      >
        <li>
          Escreva o nome do seu pokemon
        </li>
      </ul>
    </div>
  </div>
</div>
`;
